### PR TITLE
Tree Recursion

### PR DESCRIPTION
**Pull request by i-am-aisha**
## Related Issue
issue #1272 

## Proposed Changes
- Create a folder types of recursion and then further in folder make a TreeRecursion.cpp file with proper comments and explain the concepts line by line in easy to understand manner.

- Make a README.md file for types of recursion folder and list details of the TreeRecusion.cpp file like algorithm , advantages , disadvantages and  time - space complexities.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

 - [] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
  - [] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
Code image
Screenshot (5)

input image 
Screenshot (6)

Output image
Screenshot (3)

### CODE REVIEW

File: R-Recursion/Types of Recursion/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+- [Tree Recursion](#tree-recursion)
+  - [Algorithm](#algorithm)
+  - [Properties](#properties)
+  - [Advantages](#advantages)
+  - [Disadvantages](#disadvantages)
+
```
Comment (Mrjoy832) [inline R-Recursion/Types of Recursion/README.md:6]: Hope Later You will add a index with all other types of Recursion

**Comment by Mrjoy832** [comment]
image
Where are last 3 checkboxes ...add those and put tick on proper box

**Comment by i-am-aisha** [comment]
updated 👍

**Comment by Mrjoy832** [comment]
image
Do it

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

**Comment by i-am-aisha** [comment]
Done 👍

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
LGTM🌟

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🚩

**Pull request merged**